speaker/main.go

```diff
@@ -531,6 +544,19 @@ func compareIPs(ips1, ips2 []net.IP) bool {
 	return true
 }
 
+func parseExcludeNodePattern() (*v1.Node, error) {
```

Comment:
is there a reason why the pattern is represented by a full fledged node? I think this deserves its own struct (possibly with a "matches" method)

Reply:
because I would like not create extra structs/methods and leverage the k8s library. Plus I wanted to keep full fledged node  in case we need to extend in the future. E.g disable when noschedulable (which I do think it makes sense under some condition), or node special condition or taint or whatever is in the v1.Node stuct.

Nevertheless, if we agree that we strictly care only about annotation and labels, then I could drop that implementation and have custom struct with parsing and matches method. Would you like me to do that?

Reply:
Yes I think it's confusing, as a user would expect an exact match. Let's keep just a map / two maps.